merkledag/merkledag.go

```diff
@@ -201,7 +201,20 @@ func (n *dagService) GetMany(ctx context.Context, keys []*cid.Cid) <-chan *ipld.
 	return getNodesFromBG(ctx, n.Blocks, keys)
 }
 
+func dedupKeys(keys []*cid.Cid) []*cid.Cid {
```

Comment:
I think this can be done without copying the whole array to map and reverse by running a search over the rest of the array and remembering which indexes are duplicate.

IDK if it would be faster.

Reply:
The "remember which indexes are duplicate" part is exactly what we're doing when copying to the map. We could *modify* the array instead of creating a new one but then we'd be modifying an argument. We could also sort and then dedup but this is probably faster.